src/validator.py

```diff
@@ -0,0 +1,29 @@
+import sys
+import click
+import yaml
+import json
+from jsonschema import validate
+
+
+@click.command()
+@click.option("-i", "--input", "input_files", multiple=True)
+def valdiate(input_files):
+    """
+    Valdiate a list of CyanoByte documents.
+
+    Args:
+        input_files: A list of CyanoByte documents to validate.
+    """
+    # Load the JSON Schema file
+    with open("spec/cyanobyte.schema.json", "r") as schema_json:
+        schema = json.load(schema_json)
+
+    # Validate each document against the schema
+    for input_file in input_files:
```

Comment:
Should this be indented one block, so that schema will be defined?

Reply:
Most scoping in python is not local to loops / with statements. This runs fine. Can't find a concise explanation of scoping rules in python, but this is the closest reference I could find: https://mail.python.org/pipermail/python-ideas/2008-October/002109.html

Reply:
Sure, but AFAIK if the file open fails, schema won't be defined, causing this for loop to fail.

Reply:
Just double checked, if the file open fails, it will still raise a `FileNotFoundError` or `IOError` exception.